src/site/content/en/blog/file-system-access/index.md

```diff
@@ -223,47 +223,65 @@ file at a specific position, or resize the file.
-re-open the last file when the app is opened, etc. In the text editor, I store a list of the five
+re-open the last file when the app is opened, restore the previous working directory, etc.
+In the text editor, I store a list of the five
 most recent files the user has opened, making it easy to access those files again.
 
-The code example below shows storing and retrieving a file handle.
-You can [see this in action](https://filehandle-indexeddb.glitch.me/) over on Glitch
+The code example below shows storing and retrieving a file handle and a directory handle.
+You can [see this in action](https://filehandle-directoryhandle-indexeddb.glitch.me/) over on Glitch
 (I use the [idb-keyval](https://www.npmjs.com/package/idb-keyval) library for brevity).
 
 ```js
-import { get, set } from 'https://unpkg.com/idb-keyval@5.0.2/dist/esm/index.js';
+import { get, set } from "https://unpkg.com/idb-keyval@5.0.2/dist/esm/index.js";
 
-const pre = document.querySelector('pre');
-const button = document.querySelector('button');
+const pre1 = document.querySelector("pre.file");
+const pre2 = document.querySelector("pre.directory");
+const button1 = document.querySelector("button.file");
+const button2 = document.querySelector("button.directory");
 
-button.addEventListener('click', async () => {
+// File handle
+button1.addEventListener("click", async () => {
   try {
-    // Try retrieving the file handle.
-    const fileHandleOrUndefined = await get('file');    
-    if (fileHandleOrUndefined) {      
-      pre.textContent =
-          `Retrieved file handle "${fileHandleOrUndefined.name}" from IndexedDB.`;
+    const fileHandleOrUndefined = await get("file");
+    if (fileHandleOrUndefined) {
+      pre1.textContent = `Retrieved file handle "${fileHandleOrUndefined.name}" from IndexedDB.`;
       return;
     }
-    // This always returns an array, but we just need the first entry.
     const [fileHandle] = await window.showOpenFilePicker();
-    // Store the file handle.
-    await set('file', fileHandle);    
-    pre.textContent =
-        `Stored file handle for "${fileHandle.name}" in IndexedDB.`;
+    await set("file", fileHandle);
+    pre1.textContent = `Stored file handle for "${fileHandle.name}" in IndexedDB.`;
+  } catch (error) {
+    alert(error.name, error.message);
+  }
+});
+
+// Directory handle
+button2.addEventListener("click", async () => {
+  try {
+    const directoryHandleOrUndefined = await get("directory");
+    if (directoryHandleOrUndefined) {
+      pre2.textContent = `Retrieved directroy handle "${directoryHandleOrUndefined.name}" from IndexedDB.`;
+      return;
+    }
+    const directoryHandle = await window.showDirectoryPicker();
+    await set("directory", directoryHandle);
+    pre2.textContent = `Stored directory handle for "${directoryHandle.name}" in IndexedDB.`;
   } catch (error) {
     alert(error.name, error.message);
   }
 });
 ```
 
+### Stored file or directory handles and permissions
+
```

Comment:
It looks like some text below here might need to be updated as well. Line 286, for example, should be 'has granted permission to the file or directoruy...'

Reply:
Good point. Added in https://github.com/GoogleChrome/web.dev/pull/4698/commits/49dc0144e595c60b43b26991f6d526820866338d.